x-pack/plugins/cloud_security_posture/public/components/setup_status.tsx

```diff
@@ -0,0 +1,187 @@
+    })}
+    solution={i18n.translate('xpack.csp.setupStatus.packageNotInstalled.solutionNameLabel', {
+      defaultMessage: 'Cloud Security Posture',
+    })}
+    // TODO: Add real docs link once we have it
+    docsLink={'https://www.elastic.co/guide/index.html'}
+    logo={'logoSecurity'}
+    actions={{
+      elasticAgent: {
+        href: cisIntegrationLink,
+        isDisabled: !cisIntegrationLink,
+        title: i18n.translate('xpack.csp.setupStatus.packageNotInstalled.buttonLabel', {
+          defaultMessage: 'Add a CIS integration',
+        }),
+        description: i18n.translate('xpack.csp.setupStatus.packageNotInstalled.description', {
+          defaultMessage:
+            'Use our CIS Kubernetes Benchmark integration to measure your Kubernetes cluster setup against the CIS recommendations.',
+        }),
+      },
+    }}
+  />
+);
+
+const defaultLoadingRenderer = () => (
+  <CspLoadingState data-test-subj={LOADING_STATE_TEST_SUBJECT}>
+    <FormattedMessage id="xpack.csp.setupStatus.loadingDescription" defaultMessage="Loading..." />
+  </CspLoadingState>
+);
+
+const defaultErrorRenderer = (error: unknown) => (
+  <EuiEmptyPrompt
+    css={css`
+      margin-top: 50px;
+    `}
+    color="danger"
+    iconType="alert"
+    data-test-subj={ERROR_STATE_TEST_SUBJECT}
+    title={
+      <h2>
+        <FormattedMessage
+          id="xpack.csp.setupStatus.errorRenderer.errorTitle"
+          defaultMessage="We couldn't fetch your cloud security posture data"
+        />
+      </h2>
+    }
+    body={
+      isCommonError(error) ? (
+        <p>
+          <FormattedMessage
+            id="xpack.csp.setupStatus.errorRenderer.errorDescription"
+            defaultMessage="{error} {statusCode}: {body}"
+            values={{
+              error: error.body.error,
+              statusCode: error.body.statusCode,
+              body: error.body.message,
+            }}
+          />
+        </p>
+      ) : undefined
+    }
+  />
+);
+
+const defaultNoDataRenderer = () => {
+  return (
+    <NoDataPage
+      data-test-subj={DEFAULT_NO_DATA_TEST_SUBJECT}
+      css={css`
+        margin-top: 50px;
+      `}
+      pageTitle={i18n.translate('xpack.csp.setupStatus.defaultNoDataConfig.pageTitle', {
+        defaultMessage: 'No data found',
+      })}
+      solution={i18n.translate('xpack.csp.setupStatus.defaultNoDataConfig.solutionNameLabel', {
+        defaultMessage: 'Cloud Security Posture',
+      })}
+      // TODO: Add real docs link once we have it
+      docsLink={'https://www.elastic.co/guide/index.html'}
+      logo={'logoSecurity'}
+      actions={{}}
+    />
+  );
+};
+
+interface SetupStatusProps<TData, TError> {
+  children: React.ReactNode;
+  query?: UseQueryResult<TData, TError>;
+  loadingRender?: () => React.ReactNode;
+  errorRender?: (error: TError) => React.ReactNode;
+  noDataRenderer?: () => React.ReactNode;
+}
+
+export const SetupStatus = <TData, TError>({
```

Comment:
Consider renaming `SetupStatus` to something like `CspPage` to better convey the purpose of this component, which is a page wrapper with specific CSP domain knowledge that determines a render UI state for our pages by checking `setup_status` API, `query/children` and `noDataConfig` to make that decision.

Naming can sound nit picky but I think this is an important root component and `SetupStatus` doesn't fully convey everything that it does, so it might be worth nitting over.